### PR TITLE
Move docs node generation to a local plugin

### PR DESCRIPTION
**Pull request by rogermparent**
This PR is a first step in porting our docs engine into a reusable Gatsby theme. It moves all the files from our docs Gatsby Model to a new local plugin at `plugins/gatsby-theme-iterative-docs`.

From here, we can continue to move more parts of our docs engine (sidebar, components) to this plugin directory. Once we have everything moved and ensure it still works, we can turn the local plugin folder into its own node module and start work on reintegrating it with the other sites.

### CODE REVIEW

File: gatsby-config.js [hunk 1]
```diff
@@ -36,6 +36,13 @@ const plugins = [
   'gatsby-plugin-react-helmet',
   'gatsby-plugin-sitemap',
   'gatsby-plugin-twitter',
+  {
+    resolve: 'gatsby-theme-iterative-docs',
+    options: {
+      docComponent: require.resolve('./src/templates/doc.tsx'),
```
Comment (julieg18) [inline gatsby-config.js:42]: Where is the `docComponent` being used in the theme? I'm not seeing it being used anywhere 🤔

**Review by julieg18 — APPROVED** [review]
Looks great @rogermparent! Just one small question :)

File: gatsby-config.js (hunk 1 above)
Comment (rogermparent) [reply gatsby-config.js:42]: Whoops! I renamed this to `defaultTemplate` in `createPages` but didn't end up updating it here.

**Review by yathomasi — COMMENTED** [review]
Everything looks good @rogermparent .

**Pull request merged**